### PR TITLE
More coherant KrigingAlgorithm constructors 

### PR DESCRIPTION
**Pull request by sofianehaddad**
Now KrigingAlgorithm constructors are coherant comparing to the GeneralizedLinearModel ones (data, covModel, basis). It is more coherant since the basis/basisCollection is an 'optionnal' argument

We remove also the useless keepCholesky argument in ctors

### CODE REVIEW

**Comment by sofianehaddad** [comment]
@jschueller ok for api
which flages are you talking about?

**Comment by jschueller** [comment]
normalize & keepCholeskyFactor

**Comment by sofianehaddad** [comment]
keepCholesky for sure since it is useless
for normalize what do you suggest? using only ResourceMap keys?

**Comment by jschueller** [comment]
Oh right, normalize is used in the ctor, so we cannot remove it.

File: ChangeLog [hunk 1]
```diff
@@ -32,6 +32,7 @@
   * Deprecated NumericalMathFunction::GetValidConstants|GetValidFunctions|GetValidOperators
   * Renamed ComposedNumericalMathFunction to ComposedFunction
   * Renamed LinearNumericalMathFunction to LinearFunction
+  * Revert covModel and basis arguments in KrigingAlgorithm constructors
```
Comment (jschueller) [inline ChangeLog]: use "Swap", also mention removal of keepCholeskyFactor

Comment (sofianehaddad) [inline ChangeLog]: ok

**Comment by sofianehaddad** [comment]
@jschueller done

**Comment by regislebrun** [comment]
+1

**Comment by jschueller** [comment]
hum, docstring_geom test fails in master since the merge on osx but it seems unrelated

**Pull request merged**